codegen/main.go

```diff
@@ -361,7 +361,7 @@ local function on_socket_message(socket, message)
 		end
 	elseif message.match_data then
 		if socket.on_matchdata then
-			message.data = b64.decode(message.data)
+			message.match_data.data = b64.decode(message.match_data.data)
```

Comment:
A previous commit with this fix was done on the generated code and not on the template...